x/incentive/keeper/payout.go

```diff
@@ -0,0 +1,176 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/auth"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	"github.com/cosmos/cosmos-sdk/x/auth/vesting"
+	supplyExported "github.com/cosmos/cosmos-sdk/x/supply/exported"
+	"github.com/kava-labs/kava/x/incentive/types"
+	validatorvesting "github.com/kava-labs/kava/x/validator-vesting"
+)
+
+// PayoutClaim sends the timelocked claim coins to the input address
+func (k Keeper) PayoutClaim(ctx sdk.Context, addr sdk.AccAddress, denom string, id uint64) sdk.Error {
+	claim, found := k.GetClaim(ctx, addr, denom, id)
+	if !found {
+		return types.ErrClaimNotFound(k.codespace, addr, denom, id)
+	}
+	claimPeriod, found := k.GetClaimPeriod(ctx, id, denom)
+	if !found {
+		return types.ErrClaimPeriodNotFound(k.codespace, denom, id)
+	}
+	err := k.SendTimeLockedCoinsToAccount(ctx, types.IncentiveMacc, addr, sdk.NewCoins(claim.Reward), int64(claimPeriod.TimeLock.Seconds()))
+	if err != nil {
+		return err
+	}
+
+	ctx.EventManager().EmitEvent(
+		sdk.NewEvent(
+			types.EventTypeClaim,
+			sdk.NewAttribute(types.AttributeKeySender, fmt.Sprintf("%s", addr)),
+		),
+	)
+	return nil
+}
+
+// SendTimeLockedCoinsToAccount sends time-locked coins from the input module account to the recipient. If the recipients account is not a vesting account, it is converted to a periodic vesting account and the coins are added to the vesting balance as a vesting period with the input length.
+func (k Keeper) SendTimeLockedCoinsToAccount(ctx sdk.Context, senderModule string, recipientAddr sdk.AccAddress, amt sdk.Coins, length int64) sdk.Error {
+	macc := k.supplyKeeper.GetModuleAccount(ctx, senderModule)
+	if !macc.GetCoins().IsAllGTE(amt) {
+		return types.ErrInsufficientModAccountBalance(k.codespace, senderModule)
+	}
+
+	// 0. Get the account from the account keeper and do a type switch, error if it's a validator vesting account or module account (can make this work for validator vesting later if necessary)
+	acc := k.accountKeeper.GetAccount(ctx, recipientAddr)
+
+	switch acc.(type) {
+	case *validatorvesting.ValidatorVestingAccount, supplyExported.ModuleAccountI:
+		return types.ErrInvalidAccountType(k.codespace, acc)
+	case *vesting.PeriodicVestingAccount:
+		return k.SendTimeLockedCoinsToPeriodicVestingAccount(ctx, senderModule, recipientAddr, amt, length)
+	case *auth.BaseAccount:
+		return k.SendTimeLockedCoinsToBaseAccount(ctx, senderModule, recipientAddr, amt, length)
+	default:
+		return types.ErrInvalidAccountType(k.codespace, acc)
+	}
+}
+
+// SendTimeLockedCoinsToPeriodicVestingAccount sends time-locked coins from the input module account to the recipient
+func (k Keeper) SendTimeLockedCoinsToPeriodicVestingAccount(ctx sdk.Context, senderModule string, recipientAddr sdk.AccAddress, amt sdk.Coins, length int64) sdk.Error {
```

Comment:
can you split this func into multiple private ones?

Reply:
Split out the logic for updating the vesting schedule.